docs-src/go/how-to-use-worker-versioning-in-go.md

```diff
@@ -0,0 +1,122 @@
+---
+id: how-to-use-worker-versioning-in-go
+title: How to use Worker Versioning in Go
+sidebar_label: Worker Versioning
+description: Learn how to version your Go Workers by using build-ID based versioning
+tags:
+  - go
+  - how-to
+---
+
+To make use of Worker Versioning in Go, you will need to
+do the following:
+
+1. Determine and assign a Build ID to your built Worker code, and opt in to versioning
+2. Tell the Task Queue your Worker is listening on about that Build ID, and whether its compatible with an existing Build ID
+
+### Assign a Build ID to your Worker
+
+Let's say you've chosen `deadbeef` as your Build ID, which might be a short git commit hash (a
+reasonable choice as Build ID). To assign it in your Worker code, you'd assign
+the following Worker Options:
+
+```go
+// ...
+workerOptions := worker.Options{
+   BuildID: "deadbeef",
+   UseBuildIDForVersioning: true,
+// ...
+}
+w := worker.New(c, "your_task_queue_name", workerOptions)
+// ...
+```
+
+That's all you need to do in your Worker code. Importantly, if you start this Worker, it won't
+receive any tasks. That's because you need to tell the Task Queue about your Worker's Build ID
+first.
```

Comment:
```suggestion
That's all you need to do in your Worker code.
Importantly, if you start this Worker, it won't receive any tasks.
That's because you need to tell the Task Queue about your Worker's Build ID first.
```